sdk/core/Azure.Core/src/Internal/AzureBaseBuffersExtensions.cs

```diff
@@ -63,7 +59,7 @@ public static async Task WriteAsync(this Stream stream, ReadOnlySequence<byte> b
                     }
                     else
                     {
-                        if (array == null || buffer.Length < segment.Length)
+                        if (array == null || array.Length < segment.Length)
```

Comment:
We need to check the `array.Length` here since the `array` holds the existing segment in it. 'buffer.Length' is the length of the entire sequence which will never be smaller than the length of an individual segment (it can be equal).